### PR TITLE
Fix kubelet logs --follow bug

### PR DESCRIPTION
**Pull request by feiskyer**
Fix kubelet logs --follow bug #13811

### CODE REVIEW

**Comment by yujuhong** [comment]
Redirected to @lavalamp to find a reviewer.

File: pkg/apiserver/apiserver.go [hunk 1]
```diff
@@ -258,7 +258,13 @@ func write(statusCode int, apiVersion string, codec runtime.Codec, object runtim
 		w.WriteHeader(statusCode)
 		writer := w.(io.Writer)
 		if flush {
-			writer = flushwriter.Wrap(w)
+			if _, ok := w.(*hijackTimeoutWriter); ok {
```
Comment (lavalamp) [inline pkg/apiserver/apiserver.go]: This is a perfect use case for a type switch:

```go
switch innerWriter := w.(type) {
case hijackTimeoutWriter:
  writer = flushwriter.Wrap(innerWriter.w)
...
}
```

Comment (lavalamp) [inline pkg/apiserver/apiserver.go]: But more importantly, what goes wrong with these writers that you need to dig inside? Presumably they're wrapped for a reason.

Comment (feiskyer) [inline pkg/apiserver/apiserver.go]: @avalamp Thanks for advise, I have submitted new code. It could avoid type conversions.

**Comment by feiskyer** [comment]
@lavalamp Please review new commits.

**Comment by lavalamp** [comment]
@feiskyer thanks, this looks much better!

LGTM

**Comment by feiskyer** [comment]
Rebased to trigger e2e test since there is `Error fetching remote repo 'origin'`.

**Comment by feiskyer** [comment]
@lavalamp Please help with e2e test. I can't figure out what's wrong here.

**Comment by lavalamp** [comment]
Sorry about the testing. You hit what appear to be three different flakes.

**Comment by feiskyer** [comment]
@lavalamp  I see e2e testing is down at kubernetes-dev. Fortunately this PR has passed e2e testing.

**Pull request merged**